dashboard/app/controllers/api/v1/regional_partners_controller.rb

```diff
@@ -18,20 +18,36 @@ def capacity
 
   # GET /api/v1/regional_partners/find
   def find
-    if params[:school]
-      school = School.find(params[:school])
-      state = school.state
-      zip_code = school.zip
-    else
-      zip_code = params[:zip_code]
-      state = params[:state]
+    zip_code = params[:zip_code]
+    state = nil
 
-      # lookup state abbreviation, since the supplied state can be the full name
-      state = get_us_state_abbr_from_name(state, true) if state && state.length > 2
-    end
+    # Try to find the matching partner using the ZIP code.
+    partner = RegionalPartner.find_by_region(zip_code, nil)
+
+    # Otherwise, get the state for the ZIP code and try to find the matching partner using that.
+    unless partner
+      begin
+        Geocoder.with_errors do
+          # Geocoder can raise a number of errors including SocketError, with a common base of StandardError
+          # See https://github.com/alexreisner/geocoder#error-handling
+          Retryable.retryable(on: StandardError) do
+            state = Geocoder.search(zip_code)&.first&.state_code
+          end
+        end
+      rescue StandardError => e
+        # Log geocoding errors to honeybadger but don't fail
+        Honeybadger.notify(e,
+          error_message: 'Error geocoding regional partner workshop zip_code',
+          context: {
+            zip_code: zip_code
+          }
+        )
```

Comment:
Note - This means if the geocoder fails we could return nil (no result found) when really there is a result for that state. This is probably fine, and we'll be notified in honeybadger to look into it. I just want to call it out here that the user will see no regional partner found.

Reply:
Yes, I see Geocoder fails as the biggest risk to this change, but tracking with Honeybadger should help us understand if we need to change strategy.  (We could show the state dropdown in such a case, but I wanted to avoid too many unusual code paths if it's not necessary.)

Reply:
FWIW we use it in many other places and at least for the ones I pay attention to like CSF [workshop material orders](https://github.com/code-dot-org/code-dot-org/blob/staging/dashboard/app/models/pd/workshop_material_order.rb#L76), I have not seen it fail. AFAICT it's very reliable.